nengo/builder/operator.py

```diff
@@ -643,6 +643,70 @@ def step_dotinc():
         return step_dotinc
 
 
+class BsrDotInc(Operator):
+    """Increment signal Y by dot(A, X) where is a matrix in block sparse row
+    format.
+
+    Requires SciPy.
+
+    Currently, this only supports matrix-vector multiplies for compatibility
+    with NengoOCL.
+
+    Parameters
+    ----------
+    A : (k, r, c) Signal
+        The signal providing the k data blocks with r rows and c columns.
+    X : (k * c) Signal
+        The signal providing the k column vectors to multiply with.
+    Y : (k * r) Signal
+        The signal providing the k column vectors to update.
+    indices : ndarray
+        Column indices, see `scipy.sparse.bsr_matrix` for details.
+    indptr : ndarray
+        Column index pointers, see `scipy.sparse.bsr_matrix` for details.
+    reshape : bool
+        Whether to reshape the result.
+    """
+
+    def __init__(self, A, X, Y, indices, indptr, reshape, tag=None):
+        super(BsrDotInc, self).__init__(tag=tag)
+
+        if X.ndim >= 2 and any(d > 1 for d in X.shape[1:]):
+            raise BuildError("X must be a column vector")
+        if Y.ndim >= 2 and any(d > 1 for d in Y.shape[1:]):
+            raise BuildError("Y must be a column vector")
+
+        self.A = A
+        self.X = X
+        self.Y = Y
+        self.indices = indices
+        self.indptr = indptr
+        self.reshape = reshape
+        self.tag = tag
+
+        self.sets = []
+        self.incs = [Y]
+        self.reads = [A, X]
+        self.updates = []
+
+    def _descstr(self):
+        return '%s, %s -> %s' % (self.A, self.X, self.Y)
+
+    def make_step(self, signals, dt, rng):
+        X = signals[self.X]
+        A = signals[self.A]
+        Y = signals[self.Y]
+
+        def step_dotinc():
+            from scipy.sparse import bsr_matrix
+            mat_A = bsr_matrix((A, self.indices, self.indptr))
```

Comment:
For some reason my email reply does not show up here ... anyways:

I don't think that this operation is very costly because no data should be copied. But if it is possible to do the construction only once that would of course be great. I have the feeling that I tried that and for some reason it was not possible, but I cannot recall the exact details anymore. I'll think a bit more about it.